app/uk/gov/hmrc/ctcguaranteebalanceeisstub/models/GuaranteeReferenceNumber.scala

```diff
@@ -24,6 +24,10 @@ case class GuaranteeReferenceNumber(value: String) extends AnyVal {
     case GuaranteeReferenceNumber.grnPattern(_) => true
     case _                                      => false
   }
+
+  def isNotFound(): Boolean =
+    if (value.startsWith("1")) true // we consider all GRNs starting with "1" as not found in the DB.
```

Comment:
@drnaylor-oc should we not check with matching regex pattern for grn?  if matched should be true. otherwise false?

Reply:
We already do, the function above this has the code and that check is done before this one.

Reply:
so I beleive   case false => Left(InternalServerError(JsString(s"The guarantee reference number [${request.GRN.value}] is not in the correct format."))) should change to NOTFOUND

Reply:
isNotFound just means it's not in the DB - it's a stub!

Reply:
value.startsWith("1") already returns a Boolean, doesn't need to be wrapped in if statement.